src/os_posix/os_alloc.c

```diff
@@ -9,6 +9,27 @@
 #include "wt_internal.h"
 
 /*
+ * On systems with poor default allocators for allocations greater than 16 KB,
+ * we provide an option to use TCMalloc explicitly.
+ * This is important on Windows which does not have a builtin mechanism
+ * to replace C run-time memory management functions with alternatives.
+ */
+#ifdef HAVE_LIBTCMALLOC
+#include <gperftools/tcmalloc.h>
+
+/*
+ * Define HAVE_POSIX_MEMALIGN explicitly since TCMalloc supports it
+ */
+#define	HAVE_POSIX_MEMALIGN 1
```

Comment:
@markbenvenuto, the problem with this is that it won't have the right effect because other parts of the system test `HAVE_POSIX_MEMALIGN` (see `conn/conn_api.c`), and it's tied into buffer-alignment on Linux (see `build_posix/configure.ac.in`), so I'm going to remove this line.

I think we can fix it up later if there's ever any need for this configuration on a POSIX system?